plugins/item_tasks/server/rest.py

```diff
@@ -130,8 +131,25 @@ def _transformInputs(self, inputs, token):
                 if rtype not in {'file', 'item', 'folder'}:
                     raise ValidationException('Invalid input resource_type: %s.' % rtype)
 
-                resource = self.model(rtype).load(
-                    v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                try:
+                    resource = self.model(rtype).load(
+                        v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                except ValidationException:
+                    # if we asked for a file, we may have been given an item,
+                    # which case get the first file within it.  If the item's
+                    # metadata includes a dictionary that includes fileId, use
+                    # that file id.
+                    if rtype != 'file':
+                        raise
+                    item = Item().load(
+                        v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                    fileIds = [v['fileId'] for k, v in six.iteritems(item)
+                               if isinstance(v, dict) if v.get('fileId')]
+                    if len(fileIds) == 1:
+                        resource = File().load(fileIds[0], force=True)
+                    else:
+                        resource = list(Item().childFiles(
+                            item, limit=1, sort=[('_id', SortDir.ASCENDING)]))[0]
```

Comment:
I think we should prevent the `IndexError` here if the item has no files.

Reply:
If you've asked for an image and the item has no files, what should be the behavior?  Perhaps `ValidationException('Item %s has no files')` would be appropriate.